modules/build/src/main/scala/scala/build/CrossSources.scala

```diff
@@ -152,6 +152,41 @@ object CrossSources {
         .map(_.flatten)
 
     val preprocessedInputFromArgs = value(preprocessSources(inputs.flattened()))
+    val preprocessedWithUsingDirs = preprocessedInputFromArgs.filter(_.directivesLocation.isDefined)
+
+    if (preprocessedWithUsingDirs.length > 1) {
+      val projectFilePath = inputs.elements.projectSettingsFiles.headOption match
+        case Some(s) => s.path
+        case _       => inputs.workspace / "project.scala"
```

Comment:
The project settings file name `"project.scala"` should be added to `Constants` file, instead of hardcoded in code.